CoordgenConfig.hpp

```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+
+#ifdef WIN32
+#define EXPORT_COORDGEN __declspec(dllimport)
```

Comment:
This needs to be marked as import, but only if we are compiling inside of the shared library. So, you'd want to define something like

```
#ifdef IN_COORDGEN
#ifdef WIN32
#define EXPORT_COORDGEN __declspec(dllexport)
#else
#define EXPORT_COORDGEN __attribute__((visibility("default")))
#endif

#else

#ifdef WIN32
#define EXPORT_COORDGEN __declspec(dllimport)
#else
#define EXPORT_COORDGEN
#endif

#endif
```

Then add to CMakeLists:

```
target_compile_definitions(coordgenlibs PRIVATE IN_COORDGEN)
set_property(TARGET coordgenlibs PROPERTY CXX_VISIBILITY_PRESET "hidden") # used for linux / windows to mark visibility
```

If you don't mark everything in public API in EXPORT_COORDGEN, example will fail to build on OS X.